test/components/structures/ThreadPanel-test.tsx

```diff
@@ -75,4 +89,110 @@ describe('ThreadPanel', () => {
+            roomId: roomWithThreads.roomId,
+        });
+        const justBobThread = makeThread(client, roomWithThreads, {
+            authorId: bobId,
+            participantUserIds: [bobId],
+            roomId: roomWithThreads.roomId,
+        });
+        const everyoneThread = makeThread(client, roomWithThreads, {
+            authorId: charlieId,
+            participantUserIds: [aliceId, bobId, charlieId],
+            length: 5,
+            roomId: roomWithThreads.roomId,
+        });
+        roomWithThreads.threads.set(aliceAndBobThread.id, aliceAndBobThread);
+        roomWithThreads.threads.set(justBobThread.id, justBobThread);
+        roomWithThreads.threads.set(everyoneThread.id, everyoneThread);
+
+        beforeEach(() => {
+            mocked(client.getUserId).mockReturnValue(aliceId);
+            mocked(client.getCapabilities).mockResolvedValue({});
+        });
+
+        describe('when extra capabilities are not enabled on server', () => {
+            it('returns an empty timelineset when room has no threads', async () => {
+                const result = await getThreadTimelineSet(client, room);
+
+                expect(result.getLiveTimeline().getEvents()).toEqual([]);
+            });
+
+            it('returns a timelineset with thread root events for room when filter is All', async () => {
+                const result = await getThreadTimelineSet(client, roomWithThreads);
+
+                const resultEvents = result.getLiveTimeline().getEvents();
+                expect(resultEvents.length).toEqual(3);
+                expect(resultEvents).toEqual(expect.arrayContaining([
+                    justBobThread.rootEvent,
+                    aliceAndBobThread.rootEvent,
+                    everyoneThread.rootEvent,
+                ]));
+            });
+
+            it('returns a timelineset with threads user has participated in when filter is My', async () => {
+                // current user is alice
+                mocked(client).getUserId.mockReturnValue(aliceId);
+
+                const result = await getThreadTimelineSet(client, roomWithThreads, ThreadFilterType.My);
+                const resultEvents = result.getLiveTimeline().getEvents();
+                expect(resultEvents).toEqual(expect.arrayContaining([
+                    // alice authored root event
+                    aliceAndBobThread.rootEvent,
+                    // alive replied to this thread
+                    everyoneThread.rootEvent,
+                ]));
+            });
+        });
+
+        describe('when extra capabilities are enabled on server', () => {
+            beforeEach(() => {
+                jest.clearAllMocks();
+                mocked(client.getCapabilities).mockResolvedValue({
+                    ['io.element.thread']: { enabled: true },
+                });
+            });
+
+            it('creates a filter with correct definition when filterType is All', async () => {
```

Comment:
These don't test much, but mocking the next stage to get a meaningful timelineSet response to assert on is pretty hard.